### PR TITLE
Add new style for inline code

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

New style for inline code in the RTE validated by Design.

The new style is available in the RTE and in the timeline (without labs flag). Only code tag is in the scope of the PR, code blocks are not covered (should arrive soon)

I had to add the new css variables to the legacy themes, to avoid error during the CSS build.

## Style in the RTE

image

image

### Classic style layout

image

image

### Bubble layout:

image

image

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add new style for inline code ([\#9936](https://github.com/matrix-org/matrix-react-sdk/pull/9936)). Contributed by @florianduros.

### CODE REVIEW

File: res/css/views/rooms/_EventTile.pcss [hunk 1]
```diff
@@ -551,6 +551,18 @@ $left-gutter: 64px;
             background-color: $codeblock-background-color;
         }
 
+        code:not(pre *) {
+            background-color: $inlinecode-background-color;
+            border: 1px solid $inlinecode-border-color;
+            border-radius: 4px;
+            // The horizontal padding is added by gfm.css .markdown-body
+            padding: 2px 0;
```
Comment (weeman1337) [inline res/css/views/rooms/_EventTile.pcss:559]: Can you use the spacing vars everywhere? Like for instance `$spacing-2` here.

**Review by weeman1337 — APPROVED** [review]
Tested on the deployment. :+1: 

Just one smaller comment

File: res/css/views/rooms/_EventTile.pcss (hunk 1 above)
Comment (florianduros) [reply res/css/views/rooms/_EventTile.pcss:559]: Yes, good idea!

File: res/css/_spacing.pcss [hunk 2]
```diff
@@ -16,6 +16,7 @@ limitations under the License.
 
 /* 1rem :: 10px */
 
+$spacing-2: 2px;
```
Comment (richvdh) [inline res/css/_spacing.pcss:19]: I thought the whole point of the `$spacing` variables was to encourage people not to use values which are not a multiple of 4?

Comment (florianduros) [reply res/css/_spacing.pcss:19]: 1. I was not aware that we only accept multiple of 4 (the code style doesn't mention it https://github.com/vector-im/element-web/blob/develop/code_style.md#stylesheets-pcss--postcss--plugins)
2. The expected design was with a 2px spacing 🧐

Comment (richvdh) [reply res/css/_spacing.pcss:19]: I realise those things, but my question is, what is the point of the `$spacing`, if we're just going to introduce new entries whenever we need one? `$spacing-2` is a lot longer to type than `2px`. I just don't get it.

I don't mean to pick on you @florianduros - I think it's a question for the design team: @gsouquet can you give us any guidance on use the `$spacing` variables?

Comment (kerryarchibald) [reply res/css/_spacing.pcss:19]: https://vector-im.github.io/compound/?path=/story/styles-styles--spacing

**Pull request merged**